linearmodels/iv/_utility.py

```diff
@@ -78,12 +83,27 @@ class IVFormulaParser:
     The general structure of a formula is `dep ~ exog + [endog ~ instr]`
     """
 
-    def __init__(self, formula: str, data: DataFrame, eval_env: int = 2):
+    def __init__(
+        self,
+        formula: str,
+        data: DataFrame,
+        eval_env: int = 2,
+        context: LayeredMapping | None = None,
```

Comment:
nit: Context can just be `Mapping[str, Any]`. Internally we map this to a `LayeredMapping` for better isolation and attribution.

Reply:
Thanks. Changed.